File: Chapter-5-Classes-and-Objects/008-a-sample-program-to-introduce-objects-as-function-arguments.cpp
// a sample program to introduce objects as function arguments
#include <bits/stdc++.h>

using namespace std;

class time
{
    int hours;
    int minutes;

public:
    void getTime(int hour, int minute)
    {
        hours = hour;
        minutes = minute;
    }
    void showTime()
    {
        cout << hours << " hours and " << minutes << " minutes" << endl;
    }
    void sum(time, time); /// declaration with objects as arguments
};

void time::sum(time firstTime, time secondTime)
{
    minutes = firstTime.minutes + secondTime.minutes;
    hours = minutes / 60;
    minutes %= 60;
    hours += firstTime.hours + secondTime.hours;
}

int main()
{
    time one, two, three;
    one.getTime(2, 45);
    two.getTime(3, 30);

    three.sum(one, two);

    cout << "First Time : " << endl;
    one.showTime();
    cout << "Second Time : " << endl;
    two.showTime();
    cout << "Third Time : " << endl;
    three.showTime();
}

// Output
// First Time :
// 2 hours and 45 minutes
// Second Time :
// 3 hours and 30 minutes
// Third Time :
// 6 hours and 15 minutes

File: Chapter-5-Classes-and-Objects/008-a-sample-program-to-introduce-objects-as-function-arguments.py
class Time:
    def __init__(self):
        self.hours = 0
        self.minutes = 0

    def get_time(self, hour, minute):
        self.hours = hour
        self.minutes = minute

    def show_time(self):
        print(f'{self.hours} hours and {self.minutes} minutes')
    
    def add_time(self, t1, t2):
        self.minutes = t1.minutes + t2.minutes
        self.hours = self.minutes // 60
        self.minutes = self.minutes % 60
        self.hours += t1.hours + t2.hours

if __name__ == '__main__':
    one = Time()
    two = Time()
    three = Time()

    one.get_time(2, 45)
    two.get_time(3, 30)

    three.add_time(one,two)
    print("First Time: ")
    one.show_time()
    print("Second Time: ")
    two.show_time()
    print("Third Time: ")
    three.show_time()